rfcs/0018-glides-trainsheet-events.md

```diff
@@ -0,0 +1,708 @@
+
+## Short-term storage
+Events are only maintained in Kinesis for 24 hours. Some events (such as a [Trips updated][#trips-updated] which modifies the operators) can affect trips more than 24 hours in the future. Clients SHOULD maintain records internally of these future changes. Clients MUST NOT expect that reading the Kinesis stream from the trim horizon will return all events affecting the current service day. Clients MUST tolerate receiving updates for events that reference unseen past events (such as an update to an added trip that the client did not see get added, or a [`EditorsChanged`](#Editors-Changed) event that stops editing for an inspector the client did not see start editing).
+
+## Long-term storage / querying
+For future use, events will be archived to S3, using Kinesis Firehose. They can either be sent to LAMP's existing folder, or a new folder. As the stream contains potential PII (operator badge information) any long-term storage (S3, database) MUST be encrypted at rest. Only Glides environments which map to a Data Platform / LAMP environment will have their events recorded.
+
+## Schema
+All events MUST have their schema documented in the [mbta/schemas](https://github.com/mbta/schemas) GitHub repository. Schemas SHOULD use [JSON Schema](https://json-schema.org/specification.html) 2020-12 or later.
+
+## Compatibility
+Changes to the events can happen in one of two ways: backwards-compatible or backwards-incompatible. Updating the RFC is not necessary for either type of event change, but the schema repository MUST be kept up to date when the events are changed.
+
+### Backwards-compatible changes
+Backwards compatible changes to an existing event `type` MUST meet two requirements:
+- old events MUST still be meet the schema.
+- the semantics of the event MUST not change.
+
+An example of a backwards-compatible change to an existing event would be adding an optional `reason` field to [Trip added][#trip-added].
+
+Adding a new event `type` (which is not replacing an existing event `type`) is backwards-compatible.
+
+Backwards-compatible changes to an event `type` are documented by bumping the minor version in the [mbta/schemas](https://github.com/mbta/schemas) GitHub repository.
+
+### Backwards-incompatible changes
+Any other changes are backwards-incompatible. Producers can add a new event `type` with a `v2` (or `v3`, &c) suffix to make backwards-incompatible changes. Producers MUST not make backwards-incompatible changes to an existing event `type`.
+
+Producers SHOULD continue to produce versions of the event while there are still consumers who can only interpret that version. Consumers SHOULD ignore earlier versions of the same event if they can parse a later version (i.e. ignoring `v1` events if they can parse `v2`).
+
+### Consumer requirements
+
+Consumers MUST ignore event `type`s that they do not understand.
```

Comment:
question: is this the intended mechanism for consumers to discard events containing PII they don't care about?